pkg/kube/client.go

```diff
@@ -421,6 +425,20 @@ func newClientInternal(clientFactory *clientFactory, revision string, cluster cl
 		}
 	}
 	c.version = lazy.NewWithRetry(clientWithTimeout.Discovery().ServerVersion)
+	c.isOpenShift = lazy.NewWithRetry(func() (bool, error) {
+		_, s, err := clientWithTimeout.Discovery().ServerGroupsAndResources()
```

Comment:
IDK. This is what we are using in our fork. If you have any ideas, please let me know. I'll research more.

Reply:
For some other areas we detect GKE by the version (version is like `v1.24.14-gke.1200`). We already fetch the version so its 'free'.

If we do need to use discovery, is `ServerResourcesForGroupVersion` feasible to do a more specific lookup? That could be the difference between 1 API call vs 100s (https://blog.upbound.io/scaling-kubernetes-to-thousands-of-crds)

Reply:
We can't use the version, as it's similar to vanilla k8s.

Another approach (cheapest one I guess) is to have a boolean variable eg (`OPENSHIFT`) defined in the helm charts, for the istiod deployment. It could be set using helm capabilities or freely using values - in which case could be used in the openshift profile). Is that a good option? Are we trying to get rid of such env variables?

Reply:
In general just trying to get rid of feature flags that are intended to be temporary but live forever. We have a variety of env var config, like `REVISION`, that will live forever. Let me think if there is a better way. In general `platform=openshift` may be more future proof (I can imagine needing a GCP specific thing in the future, etc)

Reply:
I've switched to the new `platform.IsOpenShift()`